crates/runtime/src/instance.rs

```diff
@@ -305,6 +224,18 @@ impl Instance {
         ptr
     }
 
+    /// The same as [`Self::store()`], except instead of panicking when the
+    /// store hasn't been configured yet, it returns None.
+    #[inline]
+    pub fn optional_store(&self) -> Option<*mut dyn Store> {
```

Comment:
I've commented more about this below but I'd ideally prefer to not have this since if you're asking for the store and someone forgot to configure it that's a bug rather than something we'll gracefully want to handle.